### DISCUSSION TITLE
Stop introducing helpers which will introduce issues with code readability

### DISCUSSION THREAD

**Discussion by bielu** [umbraco/Umbraco-CMS#11256] [Features and ideas]
Lately I start looking again trough PRs in umbraco and noticed, there is more and more helpers as pr which will cause issues with code readability and I think we should stop doing that.

To lower enter level we should figure out different ways than introducing helpers which will cause issues with debugging, understanding code etc.
As example Pr which I think it is hitting that issue really hard:  
https://github.com/umbraco/Umbraco-CMS/pull/11233

Usage of tag helpers hides real code and might cause issues with debugging, also make code much less readable, as it is not really showing the logic behind, especially if we introduce for example vue or angular, as now code will be totally unreadable as we will umb-if, ng-if in same tag.

**Comment by nul800sebastiaan** [comment]
Hi @bielu and thanks for the feedback! 👍 

While I understand your concerns, I am afraid I don't have good news for you in that regard. We, as HQ members feel like they are a super friendly way to get started with Umbraco templates without having to know all the quirks of Razor, and they help hiding complexity (or as you say "real code"). Our target audience is broad and we want to include people who are just starting out, making it easier for them to get going with something basic that will mostly satisfy their needs. 

Additionally, we are not going to be the only ones building taghelpers, there's already community members thinking about making a taghelpers library. I know one of your concerns is that you will eventually take over a project from someone else and will have to deal with them. If we don't build our own, you will still run into that problem. The proverbial cat is out of the bag now and the only ones who can stop it is Microsoft if they deprecate taghelpers.😅 

On the other hand, we will be selective with the taghelpers that we put in core, they have to make sense to us and provide a significant improvement over writing raw Razor code instead. For those helpers that we don't want, we'll be encouraging people to contribute to community packages.

But we are all ears for the different ways you mention here, if you can think of something else that makes it easier to get started then we'd love to hear suggestions.

**Reply by bielu** [reply]
@nul800sebastiaan 
"We, as HQ members feel like they are a super friendly way to get started with Umbraco templates without having to know all the quirks of Razor, and they help hiding complexity (or as you say "real code"). "
Yeah but long term they will be pain in the ass, I think much better will be do better docs with showing how to start etc. not really using tag helpers as it will be on long term big tech issue if someone will be coding whole solution on that instead of controllers and models :)
